File: 프로그래머스/lv2/42860. 조이스틱/조이스틱.py
def solution(name):
    answer = 0
    
    # 기본 이동 횟수는 길이 -1
    move = len(name) -1
    
    for i, n in enumerate(name):
        answer += min(ord(n)-ord('A'), ord('Z')-ord(n)+1)
        
        # 연속된 A 찾기
        next = i+1
        while next<len(name) and name[next]=='A':
            next += 1
        
        # 기존, 연속된 A의 왼쪽 시작, 연속된 A의 오른쪽 시작
        move = min(move, 2*i + len(name) - next, i + 2*(len(name)-next))
    
    answer += move
    
    return answer

File: 프로그래머스/2/42860. 조이스틱/조이스틱.js
function solution(name) {
    let answer = 0;
    const n = name.length
    let move = n -1
    for (let i=0; i<n; i++) {
        const s = name[i]
        answer += Math.min(s.charCodeAt() - 'A'.charCodeAt(), 'Z'.charCodeAt() - s.charCodeAt() + 1)
        
        let next = i+1
        while (next < n && name[next] == 'A') {
            next += 1
        }
        
        move = Math.min(move, 2*i + n - next, i + 2*(n-next))
    }
    
    answer += move
    return answer;
}